torch/utils/_pytree.py

```diff
@@ -283,11 +304,50 @@ def register_dataclass(cls: type[Any]) -> None:
         >>> assert torch.allclose(point.y, torch.tensor(2))
 
     """
-    import torch.export
+    if not dataclasses.is_dataclass(cls):
+        assert field_names is not None, (
+            "field_names must be specified with a list of all"
+            "attribute names if cls is not a dataclass"
+        )
+    else:
+        dataclass_fields = [f.name for f in dataclasses.fields(cls)]
+        if field_names is not None:
+            assert set(field_names) == set(dataclass_fields), (
+                "field_names does not include all dataclass fields:\n"
+                f"field_names: {field_names}\n"
+                f"dataclass fields: {dataclass_fields}"
+            )
+        else:
+            field_names = [f.name for f in dataclasses.fields(cls)]
```

Comment:
```suggestion
            field_names = dataclass_fields
```

Duplicate calculation.